qinglong/DefaultTasks/bili_task_base.sh

```diff
@@ -379,8 +395,26 @@ install_dotnet() {
 # 从github获取bilitool下载地址
 get_download_url() {
     eval $invocation
-
-    tag=$1
+    # 增加业务逻辑：主要为了解决完成抓不到版本号时造成的更新错误
+    # 可配置环境变量，自定义一个初始值，然后进行下载，避免因为返回值为null而下载不了安装包
+    # bilitool: tag.txt记录的版本：null
+    # bilitool: bilitool未安装
+    # bilitool: 开始安装环境
+    # bilitool: 开始安装bilitool
+    # bilitool: 最新版本：null
+    # bilitool: 下载地址：https://ghproxy.net/https://github.com/RayWangQvQ/BiliBiliToolPro/releases/download/null/bilibili-tool-pro-vnull-linux-musl-x64.zip
+    # % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+    #                                 Dload  Upload   Total   Spent    Left  Speed
+    # 100     9  100     9    0     0     10      0 --:--:-- --:--:-- --:--:--    10
+    # Archive:  bilitool-null.zip
+    # End-of-central-directory signature not found.  Either this file is not
+    # a zipfile, or it constitutes one disk of a multi-part archive.  In the
+    # latter case the central directory and zipfile comment will be found on
+    # the last disk(s) of this archive.
+    # unzip:  cannot find zipfile directory in one of bilitool-null.zip or
+    #         bilitool-null.zip.zip, and cannot find bilitool-null.zip.ZIP, period.
+
+    tag=${BILI_STORED_TAG:-""}
     url="${github_proxy}https://github.com/RayWangQvQ/BiliBiliToolPro/releases/download/$tag/bilibili-tool-pro-v$tag-$current_os-$machine_architecture.zip"
     say "下载地址：$url"
     echo $url
```

Comment:
# 增加业务逻辑：主要为了解决完成抓不到版本号时造成的更新错误

    关于GitHub有时异常访问，会抓不到基础关键值的问题 比如tag版本号，除了用环境变量手工指定，兼容性上看，你要不和其他作者一样，使用国内的节点做个辅助来源，通过对来访者的ip判断，墙内的直接访问如gitcode的仓库，墙外的访问github，这样安装以后再更新拉库上兼容性更好。

    我个人的兼容做法是在国内节点上部署一个私有库，如果GitHub访问有异常数据就会转到我们自己的私有库上去了
    ```suggestion
        # 增加业务逻辑：主要为了解决完成抓不到版本号时造成的更新错误

        tag=${BILI_STORED_TAG:-""}
        url="${github_proxy}https://github.com/RayWangQvQ/BiliBiliToolPro/releases/download/$tag/bilibili-tool-pro-v$tag-$current_os-$machine_architecture.zip"
        say "下载地址：$url"
        echo $url
    ```